### PR TITLE
Undo `instanceof Buffer` changes

### PR DESCRIPTION
**Pull request by feross**
It turns out that it's not possible to use `instanceof Buffer` safely, like I thought.

It's not possible to use `instanceof Buffer` reliably in a browserify context because there could be multiple different copies of the 'buffer' package in use.

This previous method (checking `buf._isBuffer`) works even for Buffer instances that were created from another copy of the `buffer` package.

NOTE: It's possible to have two different "instances" of the 'buffer' package, even if the 'buffer' package appears only once in the bundled code. This can happen if you require 'buffer' in different ways, for example:

`require('buffer')` vs. `require('buffer/')` vs. using the implicit `Buffer` global.

You can confirm this by browserifying this code:

```js
console.log(require('buffer').Buffer === require('buffer/').Buffer) // will be false
```

So, for this reason, `instanceof` won't work.

Fixes: https://github.com/feross/buffer/issues/154

### CODE REVIEW

**Comment by feross** [comment]
Released as 5.0.5.

File: index.js [hunk 1]
```diff
@@ -474,8 +474,12 @@ function slowToString (encoding, start, end) {
...
 }
 
-// The property is used by the `is-buffer` npm package to detect Buffer instances
-// in Safari 5-7. Remove this eventually.
+// This property is used by `Buffer.isBuffer` (and the `is-buffer` npm package)
+// to detect a Buffer instance. It's not possible to use `instanceof Buffer`
+// reliably in a browserify context because there could be multiple different
+// copies of the 'buffer' package in use. This method works even for Buffer
+// instances that were created from another copy of the `buffer` package.
+// See: https://github.com/feross/buffer/issues/154
```
Comment (dcousens) [inline index.js]: Perfect

**Comment by dcousens** [comment]
LGTM,  nice work @feross.
I did think of this as being an issue when I saw the `instanceof`,  but didn't think anyone would realistically run into the issue since very few people explicitly `require('buffer')`.

**Comment by feross** [comment]
Live and learn!

**Pull request merged**